scipy/sparse/csgraph/_tools.pyx

```diff
@@ -502,7 +503,9 @@ def reconstruct_path(csgraph, predecessors, directed=True):
 
     sctree = csr_matrix((data, indices, indptr), shape=(N, N))
     if is_pydata_sparse:
-        sctree = pydata_sparse_cls.from_scipy_sparse(sctree)
+        sctree = pydata_sparse_cls.from_scipy_sparse(
+            sctree, fill_value=pydata_sparse_fill_value
+        )
```

Comment:
The point about new `sparse` version also applies here

Reply:
Done!